src/viadot/sources/smb.py

```diff
@@ -0,0 +1,330 @@
+        """
+        date_filter_parsed = self._parse_dates(
+            date_filter=date_filter,
+            dynamic_date_symbols=dynamic_date_symbols,
+            dynamic_date_format=dynamic_date_format,
+            dynamic_date_timezone=dynamic_date_timezone,
+        )
+
+        self._scan_directory(self.base_path, keywords, extensions, date_filter_parsed)
+        return self.found_files
+
+    def _parse_dates(
+        self,
+        date_filter: str | tuple[str, str] | None = None,
+        dynamic_date_symbols: list[str] = ["<<", ">>"],  # noqa: B006
+        dynamic_date_format: str = "%Y-%m-%d",
+        dynamic_date_timezone: str = "Europe/Warsaw",
+    ) -> pendulum.Date | tuple[pendulum.Date, pendulum.Date] | None:
+        """Parses a date or date range, supporting dynamic date symbols.
+
+        Args:
+            date_filter (str | tuple[str, str] | None):
+                - A single date string (e.g., "2024-03-03").
+                - A tuple containing exactly two date strings, 'start' and 'end' date.
+                - None, which applies no date filter.
+                Defaults to None.
+            dynamic_date_symbols (list[str]): Symbols for dynamic date handling.
+                Defaults to ["<<", ">>"].
+            dynamic_date_format (str): Format used for dynamic date parsing.
+                Defaults to "%Y-%m-%d".
+            dynamic_date_timezone (str): Timezone used for dynamic date processing.
+                Defaults to "Europe/Warsaw".
+
+        Returns:
+            pendulum.Date: If a single date is provided.
+            tuple[pendulum.Date, pendulum.Date]: If a date range is provided.
+            None: If `date_filter` is None.
+
+        Raises:
+            ValueError: If `date_filter` is neither a string nor a tuple of exactly
+                two strings.
+        """
+        if date_filter is None:
+            return None
+
+        ddh = DynamicDateHandler(
+            dynamic_date_symbols=dynamic_date_symbols,
+            dynamic_date_format=dynamic_date_format,
+            dynamic_date_timezone=dynamic_date_timezone,
+        )
+
+        match date_filter:
+            case str():
+                return pendulum.parse(ddh.process_dates(date_filter)).date()
+
+            case (start, end) if isinstance(start, str) and isinstance(end, str):
+                return (
+                    pendulum.parse(ddh.process_dates(start)).date(),
+                    pendulum.parse(ddh.process_dates(end)).date(),
+                )
+
+            case _:
+                msg = (
+                    "date_filter must be a string, a tuple of exactly 2 dates, or None."
+                )
+                raise ValueError(msg)
+
+    def _scan_directory(
+        self,
+        path: str,
+        keywords: list[str] | None,
+        extensions: list[str] | None,
+        date_filter_parsed: pendulum.Date
+        | tuple[pendulum.Date, pendulum.Date]
+        | None = None,
+    ) -> None:
```

Comment:
This should output the results rather than storing them in an attribute

Reply:
[25aec0a](https://github.com/dyvenia/viadot/pull/1127/commits/25aec0a5435a23d563480f75a16904f79630c2b3)

Reply:
Hm this still does `self.found_files.update(found_file)` and only returns the last found file I think